src/decide.ts

```diff
@@ -102,6 +104,45 @@ export function compareFileTreesIgnoringNames(
 	);
 }
 
+export function compareFileTreesPartialIgnoringNames(
+	candidate: Metafile,
+	searchee: Searchee
+): boolean {
+	const { fuzzySizeThreshold } = getRuntimeConfig();
+	let matchedSizes = 0;
+	for (const candidateFile of candidate.files) {
+		let matchedSearcheeFiles = searchee.files.filter(
+			(searcheeFile) => searcheeFile.length === candidateFile.length
+		);
+		if (matchedSearcheeFiles.length) {
+			matchedSizes += candidateFile.length;
+		}
+	}
+	return matchedSizes / candidate.length >= 1 - fuzzySizeThreshold;
+}
+
```

Comment:
I know my earlier comments asked for a certain code architecture for `assessCandidateHelper`, but I think your initial intuition was better. would you mind changing it back? Sorry for the churn!

Reply:
This function was added to distinguish between a SIZE_MISMATCH and FILE_TREE_MISMATCH in partial. I think it makes sense for when the file sizes match but there are duplicates sizes with wrong names. The only other way to do this is to return a decision rather than a bool. The code structure can be preserved if I remove reporting a FILE_TREE_MISMATCH for partial (which should be very rare).

Do you want to keep this behavior or remove it?